src/lib/es2020.symbol.wellknown.d.ts

```diff
@@ -21,3 +21,12 @@ interface RegExp {
      */
     [Symbol.matchAll](str: string): RegExpStringIterator<RegExpMatchArray>;
 }
+
+interface String {
+    /**
+     * Matches a string or an object that supports being matched against, and
+     * returns an iterable of matches containing the results of that search.
+     * @param regexp An object that supports being matched against.
+     */
+    matchAll(matcher: { [Symbol.matchAll](str: string): RegExpStringIterator<RegExpMatchArray>; }): RegExpStringIterator<RegExpExecArray>;

```

Comment:
`String.prototype.matchAll`, when on the `[Symbol.matchAll]` codepath, just forwards the call to the passed `matcher` and returns whatever it returns.  There's no limitations on the type of the return and no coercion done.  Should this then be generic in the return type?

```
matchAll<T>(matcher: { [Symbol.matchAll](str: string): T; }): T;
```

Reply:
Wouldn't this apply to all the other `[Symbol]` RegExp symbols. I don't think it was done for them, it would also make it inconvenient to call, as in the common case, its supposed to return `RegExpStringIterator<RegExpMatchArray>`, returning something else is technically unexpected, and kinda violates what the symbol is meant to do.